7657 Implement saving EcoNews to favorites for further filtering by favorites

RockTF opened the pull request:
[7657 Implement saving EcoNews to favorites for further filtering by favorites](https://github.com/ita-social-projects/GreenCity/issues/7657) - Implement methods to add and remove **`Eco News to/from user favorites`**. Update **`EcoNews`** and **`User`** entities with necessary mappings, and ensure relevant tests and controller methods are included.

RockTF committed:
Add favorite functionality for Eco News

Implement methods to add and remove Eco News to/from user favorites. Update EcoNews and User entities with necessary mappings, and ensure relevant tests and controller methods are included.

RockTF committed:
Add update method to EcoNewsService interface

Introduce a missing `update` method to the EcoNewsService interface to synchronize it with the implementation. This addition ensures method availability and proper interface behavior across the service layer.

RockTF committed:
Fix typo in constant error message for EcoNews ID

Corrected the spelling error in the error message constant from "ECO_NEWS" to "ECO_NEW" in various parts of the codebase. Ensured consistent usage across service implementation and test files.

RockTF committed:
Add SuperBuilder to DTO classes and refactor code

Enhanced multiple DTO classes by adding the @SuperBuilder annotation for better inheritance support. Also refactored a method in the ModelUtils class to use the builder pattern, enhancing code readability and maintainability.

RockTF committed:
Refactor indentation in ModelUtils.java

Adjusted the indentation in the getAchievementManagementDto method to improve code readability and maintain consistency with the rest of the codebase. These changes help ensure uniformity across the project's code style.

RockTF committed:
Merge branch 'dev' into 7657-favorites-ecoNews

RockTF committed:
Refactor TagTranslationDto to remove redundant imports

Simplify the constructor's signature and remove unused imports in TagTranslationDto.java. Updated the test utilities in ModelUtils.java to conform with the refactored builder pattern usage.

RockTF committed:
Merge remote-tracking branch 'origin/7657-favorites-ecoNews' into 7657-favorites-ecoNews

RockTF committed:
Refactor and enhance exception handling and data classes

Added parameterized constructor to InvalidURLException for better exception details. Implemented getEmail method in SendHabitNotification and ScheduledEmailMessage to fulfill interface contract. Refactored user DTO classes to use Lombok @Data annotation for cleaner and more concise code.

RockTF committed:
Update spring-context version and fix code indentation

Updated spring-context dependency to version 6.1.14 in the service-api/pom.xml file. Also corrected the indentation for consistency in the ModelUtils.java file.

RockTF committed:
Add Lombok annotations to DTO classes

Enhanced `UserManagementDto`, `AchievementManagementDto`, and `UserManagementUpdateDto` with explicit Lombok annotations such as `Data`, `Builder`, `AllArgsConstructor`, and `NoArgsConstructor`. Also removed the version specification for `spring-context` dependency in `pom.xml` to allow for more flexible dependency management.

RockTF committed:
Add Builder.Default annotation to EcoNews entity

Added the `@Builder.Default` annotation to the `usersDislikedNews` field in the `EcoNews` entity to ensure proper default initialization when using Lombok's builder pattern. This change ensures the `usersDislikedNews` set is always initialized to avoid potential `NullPointerExceptions`.

ABbondar commented on service/src/test/java/greencity/service/EcoNewsServiceImplTest.java:
```diff
@@ -833,4 +836,99 @@ void setHiddenWithWrongIdValueThrowExceptionTest() {
+        User user = ModelUtils.getUser();
+
+        when(ecoNewsRepo.findById(1L)).thenReturn(Optional.of(ecoNews));
+        when(userRepo.findByEmail(TestConst.EMAIL)).thenReturn(Optional.of(user));
```
please, add verify to each when()

ABbondar commented on service/src/test/java/greencity/service/EcoNewsServiceImplTest.java:
```diff
@@ -833,4 +836,99 @@ void setHiddenWithWrongIdValueThrowExceptionTest() {
+        NotFoundException exception = assertThrows(NotFoundException.class, () ->
+                ecoNewsService.addToFavorites(1L, TestConst.EMAIL));
+
+        assertEquals("Eco new doesn't exist by this id: 1", exception.getMessage());
```
messages can be moved to ErrorMessage class

RockTF committed:
Improve error handling and add repository checks

Updated error messages to use constants and added verification checks for ecoNewsRepo and userRepo to ensure proper handling in EcoNewsServiceImplTest. These changes enhance code maintainability and improve the accuracy of the unit tests.

RockTF committed:
Fix inconsistent indentation in EcoNewsServiceImplTest

Aligned the 'assertThrows' method's indentation for 'addToFavorites' and 'removeFromFavorites' test cases. This ensures consistent code formatting and improves readability in the test file.

RockTF committed:
Fix inconsistent indentation in EcoNewsServiceImplTest

Aligned the 'assertThrows' method's indentation for 'addToFavorites' and 'removeFromFavorites' test cases. This ensures consistent code formatting and improves readability in the test file.

RockTF committed:
Refactor getFavorites endpoint for improved readability

Standardized formatting and corrected indentation in getFavorites endpoint. The changes enhance code readability and maintain consistency across the EcoNewsController and its corresponding tests.

RockTF committed:
Remove redundant blank line in EventController

Eliminated unnecessary blank line in the 'getAllEvents' method for better code readability and consistency. This change has no functional impact and purely aims to enhance code cleanliness.
